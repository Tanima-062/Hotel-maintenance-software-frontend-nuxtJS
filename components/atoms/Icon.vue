<template>
  <div :class="styles">
    <svg
      class="w-7 h-7 fill-current"
      :class="`${color}`"
      :style="`width: ${iconWidth}px; height: ${iconHeight}px`"
      :viewBox="`0 0 24 24`"
    >
      <path :d="icon" />
    </svg>
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: '' })
  icon!: string

  @Prop({ default: 'text-white' })
  color!: string

  @Prop({ default: '' })
  bgColor!: string

  @Prop({ default: false })
  round!: boolean

  @Prop({ default: 24 })
  iconWidth!: number

  @Prop({ default: 24 })
  iconHeight!: number

  @Prop({ default: '' })
  extraClass!: string

  get styles(): string {
    const classList = []
    if (this.bgColor) {
      classList.push(`${this.bgColor}`)
    }
    if (this.round) {
      classList.push('rounded-full')
    }
    if (this.extraClass) {
      classList.push(this.extraClass)
    }
    return classList.join(' ')
  }
}
</script>
