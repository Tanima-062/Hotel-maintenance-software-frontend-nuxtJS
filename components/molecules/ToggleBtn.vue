<template>
  <label :class="toggleStyle" @click="click">
    <span v-if="isActive" :class="onTextStyle">{{ toggleMenu[0].name }}</span>
    <div
      class="w-5 h-5 flex justify-center items-center rounded-full shadow-lg bg-white"
    >
      <Icon :icon="icon.blank" aria-hidden="true" />
    </div>
    <span v-if="!isActive" :class="offTextStyle">{{ toggleMenu[1].name }}</span>
  </label>
</template>

<script lang="ts">
import { Component, Vue, Prop, Emit } from 'vue-property-decorator'
import { mdiCheckboxBlankCircle, mdiCheckboxBlankCircleOutline } from '@mdi/js'
import { Global } from '~/types'

@Component({
  components: {
    Icon: () => import('~/components/atoms/Icon.vue'),
  },
})
export default class extends Vue {
  @Prop()
  toggleMenu!: Global.ToggleBtn[]

  @Prop({ default: false })
  isActive!: boolean

  private icon: { [key: string]: string } = {
    blank: mdiCheckboxBlankCircle,
    white: mdiCheckboxBlankCircleOutline,
  }

  get toggleStyle() {
    return [
      'h-8 w-min flex items-center px-1',
      'rounded-full',
      'overflow-hidden cursor-pointer',
      this.isActive ? 'bg-menuColor-active' : 'bg-btnColor-border',
      'transition-bg',
    ]
  }

  get onTextStyle() {
    return [
      'ml-4 mr-2 text-white overflow-hidden whitespace-no-wrap',
      this.isActive ? '' : 'hidden',
    ]
  }

  get offTextStyle() {
    return [
      'ml-2 mr-4 text-white overflow-hidden  whitespace-no-wrap',
      this.isActive ? 'hidden' : '',
    ]
  }

  @Emit()
  private click(): boolean {
    return !this.isActive
  }
}
</script>
<style lang="sass" scoped>
.transition-bg
  transition: background .4s ease
</style>
