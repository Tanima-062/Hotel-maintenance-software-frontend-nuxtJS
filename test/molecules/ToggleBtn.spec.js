import { mount } from '@vue/test-utils'
import ToggleBtn from '@/components/molecules/ToggleBtn.vue'
import Icon from '@/components/atoms/Icon.vue'

describe('Molecules ToggleBtn Check', () => {
  test('ToggleBtn True Test', () => {
    const props = {
      toggleMenu: [
        {
          name: 'test 1',
          value: 1,
        },
        {
          name: 'test 2',
          value: 2,
        },
      ],
      isActive: true,
    }
    const wrapper = mount(ToggleBtn, { propsData: props, stubs: { Icon } })
    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })

  test('ToggleBtn False Test', () => {
    const props = {
      toggleMenu: [
        {
          name: 'test 1',
          value: 1,
        },
        {
          name: 'test 2',
          value: 2,
        },
      ],
      isActive: false,
    }
    const wrapper = mount(ToggleBtn, { propsData: props, stubs: { Icon } })
    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
