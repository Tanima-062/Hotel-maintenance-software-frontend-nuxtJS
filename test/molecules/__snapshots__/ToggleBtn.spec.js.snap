// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules ToggleBtn Check ToggleBtn False Test 1`] = `
<label
  class="h-8 w-min flex items-center px-1 rounded-full overflow-hidden cursor-pointer bg-btnColor-border transition-bg"
>
  <!---->
   
  <div
    class="w-5 h-5 flex justify-center items-center rounded-full shadow-lg bg-white"
  >
    <div
      aria-hidden="true"
      class=""
    >
      <svg
        class="w-7 h-7 fill-current text-white"
        style="width: 24px; height: 24px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        />
      </svg>
    </div>
  </div>
   
  <span
    class="ml-2 mr-4 text-white overflow-hidden  whitespace-no-wrap"
  >
    test 2
  </span>
</label>
`;

exports[`Molecules ToggleBtn Check ToggleBtn True Test 1`] = `
<label
  class="h-8 w-min flex items-center px-1 rounded-full overflow-hidden cursor-pointer bg-menuColor-active transition-bg"
>
  <span
    class="ml-4 mr-2 text-white overflow-hidden whitespace-no-wrap"
  >
    test 1
  </span>
   
  <div
    class="w-5 h-5 flex justify-center items-center rounded-full shadow-lg bg-white"
  >
    <div
      aria-hidden="true"
      class=""
    >
      <svg
        class="w-7 h-7 fill-current text-white"
        style="width: 24px; height: 24px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        />
      </svg>
    </div>
  </div>
   
  <!---->
</label>
`;
